src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/HostDiagnosticAnalyzerProvider.cs

```diff
@@ -3,15 +3,64 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Immutable;
+using System.Reflection;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Workspaces.ProjectSystem;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
 internal class HostDiagnosticAnalyzerProvider : IHostDiagnosticAnalyzerProvider
 {
+    /// <summary>
+    /// The full path to the Razor source generator
+    /// </summary>
+    public string? RazorSourceGenerator { get; set; }
+
+    private ImmutableArray<(AnalyzerFileReference reference, string extensionId)>? _cachedAnalyzerReferences;
```

Comment:
Instead of making this nullable, consider just making it a Lazy<>.  Then you're guaranteed to only initialze it once.

Reply:
Does it even need to be lazy? Seems like it could just be created in the constructor, based on the parameter being non-null, and be done with it?

```
if (razorSourceGenerator is null)
{
    _analyzerReferences = Empy;
}
else
{
    _analyzerReferences = ImmutableArray.Create(new AnalyzerFileReference(razorSourceGenerator, new SimpleAnalyzerAssemblyLoader()));`
}
```

Reply:
Yeah, probably could be